File: colt-steele/minSubArrayLen.rb
def min_sub_array_len(arr, sum)
    total = 0
    start = 0
    finish = 0
    minLen = Float::INFINITY

    while start < arr.length do
        
        if total < sum && finish < arr.length
            total += arr[finish]
            finish += 1
        elsif total >= sum 
            minLen = [minLen, (finish - start)].min
            total -= arr[start]
            start += 1
        else
            break
        end
    end
    minLen === Float::INFINITY ? 0 : minLen
end 

puts min_sub_array_len([1, 4, 16, 22, 5, 7, 8, 9, 10], 39)

File: colt-steele/minSubArrayLen.js
// Arguments: array of integers and a positive integer
// Return the minimum continuous subarray that is greater than or equal to the sum given. 
// If none exists return 0

// Attempt 2
function minSubArrayLen(arr, sum) {
    let total = 0;
    let start = 0;
    let end = 0;
    let minLen = Infinity;
    
    while (start < arr.length) {
        
        if (total < sum && end < arr.length) {
            total += arr[end];
            end ++;
        } else if (total >= sum) {
            minLen = Math.min(minLen, end-start);
            total -= arr[start];
            start++;
        } else {
            break;
        }
    }
   return minLen === Infinity ? 0 : minLen;
}